app/src/main/java/com/eventyay/organizer/core/role/update/UpdateRoleViewModel.java

```diff
@@ -0,0 +1,100 @@
+package com.eventyay.organizer.core.role.update;
+
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.ViewModel;
+
+import com.eventyay.organizer.common.ContextManager;
+import com.eventyay.organizer.common.livedata.SingleEventLiveData;
+import com.eventyay.organizer.data.event.Event;
+import com.eventyay.organizer.data.role.Role;
+import com.eventyay.organizer.data.role.RoleInvite;
+import com.eventyay.organizer.data.role.RoleRepository;
+import com.eventyay.organizer.utils.ErrorUtils;
+
+import javax.inject.Inject;
+
+import io.reactivex.disposables.CompositeDisposable;
+
+public class UpdateRoleViewModel extends ViewModel {
+
+    private final RoleRepository roleRepository;
+    private RoleInvite roleData = new RoleInvite();
+    private Role role = new Role();
+
+    @Inject
+    public UpdateRoleViewModel(RoleRepository roleRepository) {
+        this.roleRepository = roleRepository;
+    }
+
+    private final CompositeDisposable compositeDisposable = new CompositeDisposable();
+    private final SingleEventLiveData<Boolean> progress = new SingleEventLiveData<>();
+    private final SingleEventLiveData<String> error = new SingleEventLiveData<>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/fossasia/open-event-orga-app/pullRequest?prid=4887619)